tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/realworld/MultipleSingletonPluginTest.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) 2016 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.tests.realworld;
+
+import java.io.File;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.n4js.n4mf.ui.internal.N4MFActivator;
+import org.eclipse.n4js.regex.ui.internal.RegularExpressionActivator;
+import org.eclipse.n4js.tester.TesterModule;
+import org.eclipse.n4js.tests.builder.AbstractBuilderParticipantTest;
+import org.eclipse.n4js.tests.util.ProjectTestsUtils;
+import org.eclipse.n4js.ts.ui.internal.TypesActivator;
+import org.eclipse.n4js.ui.utils.N4JSInjectorSupplier;
+import org.eclipse.n4js.utils.InjectorCollector;
+import org.eclipse.xtext.ui.shared.contribution.ISharedStateContributionRegistry;
+import org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.inject.Binding;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Singleton;
+import com.google.inject.TypeLiteral;
+
+/**
+ * This test detects multiple instances of injected classes marked with @Singleton.
+ * <p>
+ * Note that instances are created lazily. Consequently, this test cannot guarantee completeness. In case a case of
+ * multiple singleton instances is missing, this test should be adjusted to provoke the lazy creation of this missing
+ * case.
+ * <p>
+ * Note also that singleton classes that are never bound explicitly (meaning in none of the modules), are not checked in
+ * this test.
+ */
+public class MultipleSingletonPluginTest extends AbstractBuilderParticipantTest {
```

Comment:
Maybe use `AbstractPluginUITest` as a base. This test is for the whole product and not only builder. In this particular case, it might not change anything. Matter of taste?

Reply:
I set up and tear down the external libraries because this should trigger some lazy loading of some more singletons. However, the `AbstractPluginUITest` does not provide the set up and tear down methods to do so.